client/spec.yaml

```diff
@@ -0,0 +1,286 @@
+      responses:
+        "2XX":
+          description: instance query result for a tenant
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/QueryResponse"
+        "4XX":
+          description: Bad request
+        "5XX":
+          description: Server side error
+  /api/metrics/v1/{tenant}/api/v1/query_range:
+    parameters:
+      - $ref: "#/components/parameters/tenant"
+      - $ref: "#/components/parameters/query"
+      - $ref: "#/components/parameters/startTS"
+      - $ref: "#/components/parameters/endTS"
+      - $ref: "#/components/parameters/queryTimeout"
+      - in: query
+        name: step
+        description: Query resolution step width
+        schema:
+          type: string
+    get:
+      tags:
+        - queryv1
+      summary: Evaluate range queries for a tenant
+      operationId: getRangeQuery
+      description: |
+        You can evaluate range queries for a tenant
+      responses:
+        "2XX":
+          description: range query result for a tenant
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/QueryRangeResponse"
+        "4XX":
+          description: Bad request
+        "5XX":
+          description: Server side error
+  /api/metrics/v1/{tenant}/api/v1/series:
+    parameters:
+      - $ref: "#/components/parameters/tenant"
+      - $ref: "#/components/parameters/seriesMatcher"
+      # While matchers are required, in case of Observatorium, due to enforcing
+      # of tenant matchers, that requirement is fulfilled, so match[] is optional here.
+      - $ref: "#/components/parameters/startTS"
+      - $ref: "#/components/parameters/endTS"
+    get:
+      tags:
+        - seriesv1
+      summary: Get series for a tenant
+      operationId: getSeries
+      description: |
+        You can get series for a tenant
+      responses:
+        "2XX":
+          description: series for a tenant
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SeriesResponse"
+        "4XX":
+          description: Bad request
+        "5XX":
+          description: Server side error
+components:
+  parameters:
+    tenant:
+      $ref: ./parameters.yaml#/components/parameters/tenant
+    seriesMatcher:
+      $ref: ./parameters.yaml#/components/parameters/seriesMatcher
+    startTS:
+      $ref: ./parameters.yaml#/components/parameters/startTS
+    endTS:
+      $ref: ./parameters.yaml#/components/parameters/endTS
+    query:
+      $ref: ./parameters.yaml#/components/parameters/query
+    queryTimeout:
+      $ref: ./parameters.yaml#/components/parameters/queryTimeout
+  schemas:
+    # Responses from API.
+    SeriesResponse:
+      $ref: ./responses/responses.yaml#/components/schemas/SeriesResponse
+    QueryResponse:
+      $ref: ./responses/responses.yaml#/components/schemas/QueryResponse
```

Comment:
right now, the path definitions have refs that point down here and these schemas have refs that point elsewhere.
why are we doing double refs here instead of putting all of these refs directly above where they are actually used?

Reply:
Good catch! Let me try and clean this up a bit! :)

Reply:
Thanks for pointing this out @squat! Most of the imports weren't required or could be done directly, so I cleaned it up and simplified it further in https://github.com/observatorium/api/pull/270/commits/e696f2bff00ed8880766382774a7dbce9c14074c. 🙂